educator/src/Dscp/DB/SQLite/Util.hs

```diff
@@ -56,19 +56,13 @@ data RelationType
     | MxM  -- ^ Many-to-many
 
 -- | Table which stores just a relation between two tables.
-data RelationT (t :: RelationType) a b f = PrimaryKey a f :-: PrimaryKey b f
-    deriving (Generic)
+data RelationT (t :: RelationType) a b f =
+    (:-:)
+    { -- records are needed for better names in SQL schema
+      _relationL :: PrimaryKey a f
+    , _relationR :: PrimaryKey b f
```

Comment:
Frankly, I don't think that this improve field naming a lot, it only complicates the definition of `RelationT`